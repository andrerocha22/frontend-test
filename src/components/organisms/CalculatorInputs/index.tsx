/* eslint-disable @typescript-eslint/no-var-requires */
import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import { InputCurrency } from '../../atoms/input';
import { LabelText } from '../../atoms/typography';
import DateInput from '../../molecules/dateInput';
const arrowIconLeft = require('../../../assets/icons/arrow-left.svg') as string;
const arrowIconRight = require('../../../assets/icons/arrow-right.svg') as string;
const $Icon = require('../../../assets/icons/$.svg') as string;

interface CalculatorInputsProps {
  callback: (months: number, depositValue: string, amount: string) => void;
}

const CalculatorContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 1.6rem;
  grid-auto-flow: column;
  width: 100%;
  margin-top: 4rem;

  @media all and (max-width: 768px) {
    display: flex;
    flex-direction: column;
    width: 100%;
  }
`;

const InputsContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

const CalculatorInputs = ({ callback }: CalculatorInputsProps) => {
  const [numberOfDeposits, setNumberOfDeposits] = useState<number>(1);
  const [amount, setAmount] = useState<string>('0');

  const depositValue = () => {
    const valueToString = amount.toString().replace(/,/g, '');
    const depositValue = Number(valueToString) / numberOfDeposits;
    return depositValue.toFixed(2);
  };

  useEffect(() => {
    callback(numberOfDeposits, depositValue(), amount);
  }, [amount, numberOfDeposits]);

  const handleMonthDiff = (numberOfMonthlyDeposits: number) => {
    setNumberOfDeposits(numberOfMonthlyDeposits);
  };

  const handleAmountInput = (amount: string) => {
    setAmount(amount);
  };

  return (
    <CalculatorContainer data-test="component-calculator-input">
      <InputsContainer>
        <LabelText>Total Amount</LabelText>
        <InputCurrency
          icon={$Icon}
          backgroundColor={'#F4F8FA'}
          onChangeCallback={handleAmountInput}
        />
      </InputsContainer>
      <InputsContainer>
        <LabelText>Reach goal by</LabelText>
        <DateInput
          iconSrcRight={arrowIconRight}
          iconSrcLeft={arrowIconLeft}
          diffCallback={handleMonthDiff}
        />
      </InputsContainer>
    </CalculatorContainer>
  );
};

export default CalculatorInputs;
