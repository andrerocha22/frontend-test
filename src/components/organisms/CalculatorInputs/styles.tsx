import styled from 'styled-components';

export const CalculatorContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 1.6rem;
  grid-auto-flow: column;
  width: 100%;

  @media all and (max-width: 768px) {
    display: flex;
    flex-direction: column;
    width: 100%;
  }
`;

export const InputsContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;
