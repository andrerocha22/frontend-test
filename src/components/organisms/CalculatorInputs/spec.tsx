import React from 'react';
import { mount, shallow } from 'enzyme';
import { findByTestAttr } from '../../../test/testUtils';
import 'jest-styled-components';

import CalculatorInputs from './index';

const mockCallback = jest.fn();

describe('test calculator input', () => {
  const setup = (props = {}) => {
    const setupProps = { ...props };
    return shallow(
      <CalculatorInputs callback={mockCallback} {...setupProps} />
    ).dive();
  };

  //   const setupMount = (props = {}) => {
  //     const setupProps = { ...props };
  //     return mount(<CalculatorInputs callback={mockCallback} {...setupProps} />);
  //   };

  test('renders without error', () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, 'component-calculator-input');
    expect(component.length).toBe(1);
  });

  test('calculator input snapshot test ', () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, 'component-calculator-input');
    expect(component).toMatchSnapshot();
  });
});
