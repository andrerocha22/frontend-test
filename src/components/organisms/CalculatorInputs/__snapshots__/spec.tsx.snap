// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test calculator input calculator input snapshot test  1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(2,1fr);
  grid-gap: 1.6rem;
  grid-auto-flow: column;
  width: 100%;
  margin-top: 4rem;
}

@media all and (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    width: 100%;
  }
}

<div
  className="c0"
  data-test="component-calculator-input"
>
  <styled.div>
    <styled.p>
      Total Amount
    </styled.p>
    <InputCurrency
      backgroundColor="#F4F8FA"
      icon="test-file-stub"
      onChangeCallback={[Function]}
    />
  </styled.div>
  <styled.div>
    <styled.p>
      Reach goal by
    </styled.p>
    <DateInput
      diffCallback={[Function]}
      iconSrcLeft="test-file-stub"
      iconSrcRight="test-file-stub"
    />
  </styled.div>
</div>
`;
